### PR TITLE
skip charts directories when finding images

### PR DESCRIPTION
**Pull request by jeffreygolden**
#### What type of PR is this?

kind/bug

#### What this PR does / why we need it:
KOTS searches for private images in yaml files and writes Kustomization to use the Replicated registry.  When searching for private images, KOTS will search inside of subchart directories and report those images up to the parent chart's Kustomize file.  This resulted in kustomize errors complaining that the image's manifest could not be found, because it belongs to the subchart.

This PR skips searching for images in any directory named "charts", which is used to store subcharts.  The subcharts are still searched independently for private images and each subchart gets it's own Kustomize file to re-write the private image name.

#### Does this PR introduce a user-facing change?

```release-note
Fixed which private images Kustomize targets inside of Helm charts when sub-charts are present.
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: pkg/image/builder.go [hunk 1]
```diff
@@ -101,6 +101,13 @@ func GetPrivateImages(upstreamDir string, checkedImages map[string]ImageInfo, al
 			if err != nil {
 				return err
 			}
+
+			// Skip charts folders containing subcharts.  The kustomization for subcharts will be defined
+			// inside of those charts, and should not be referenced at the parent chart level.
+			if info.IsDir() && info.Name() == "charts" {
+				return filepath.SkipDir
+			}
+
```
Comment (jala-dx) [inline pkg/image/builder.go:110]: can you please add a testcase for subcharts.

**Review by divolgin — APPROVED** [review]

**Review by jala-dx — APPROVED** [review]

File: pkg/image/builder.go (hunk 1 above)
Comment (jeffreygolden) [reply pkg/image/builder.go:110]: I will add one to the next PR, we ran into a separate bug with sub-sub-charts.  If I add it to tests now the tests will fail.

File: pkg/image/builder.go [hunk 2]
```diff
@@ -101,6 +101,13 @@ func GetPrivateImages(upstreamDir string, checkedImages map[string]ImageInfo, al
 			if err != nil {
 				return err
 			}
+
+			// Skip charts folders containing subcharts.  The kustomization for subcharts will be defined
+			// inside of those charts, and should not be referenced at the parent chart level.
+			if info.IsDir() && info.Name() == "charts" {
```
Comment (jala-dx) [inline pkg/image/builder.go:107]: Can you check if this is needed only if useHelmInstall flag is set?

if info.IsDir() && info.Name() == "charts"  && (useHelmInstall[info.Name()] == true) {

Comment (jeffreygolden) [reply pkg/image/builder.go:107]: The check below this one is `if info.IsDir() && (useHelmInstall[info.Name()] == true)` , but it wasn't excluding subchart folders correctly.  Adding the useHelmInstall check would replicate the existing behavior.

**Comment by erquhart** [comment]
@divolgin will take a look here.

**Comment by jeffreygolden** [comment]
@divolgin  pinging for a re-review since this changed significantly since your first approval

**Review by divolgin — APPROVED** [review]

**Pull request merged**